clippy_lints/src/format.rs

```diff
@@ -53,14 +53,20 @@ impl<'a, 'tcx> LateLintPass<'a, 'tcx> for Pass {
                         // and that the argument is a string
                         if check_arg_is_display(cx, &args[1]);
                         then {
-                            span_lint(cx, USELESS_FORMAT, span, "useless use of `format!`");
+                            let sugg = format!("{}.to_string()", snippet(cx, args[1].span, "<expr>").into_owned());
```

Comment:
This needs to use `expr.span`, otherwise you end up with `format!("foo".to_string())`